lxd/api_project.go

```diff
@@ -1457,6 +1457,38 @@ func projectValidateConfig(s *state.State, config map[string]string, projectName
 		return fmt.Errorf("Failed loading storage pool names: %w", err)
 	}
 
+	// Per-network project limits for uplink IPs only make sense for projects with their own networks.
+	if shared.IsTrue(config["features.networks"]) {
+		// Get networks that are allowed to be used as uplinks by this project.
+		allowedUplinkNetworks, err := network.AllowedUplinkNetworks(s, config)
+		if err != nil {
+			return nil
+		}
+
+		// Add network-specific config keys.
+		for _, networkName := range allowedUplinkNetworks {
+			// lxdmeta:generate(entities=project; group=limits; key=limits.networks.uplink_ips.ipv4.NETWORK_NAME)
+			// This represents the maximum value for IPV4 addresses available on a network
+			// named NETWORK_NAME to be assigned as uplink addresses for entities inside
+			// a specific project.
+			//
+			// ---
+			//  type: string
+			//  shortdesc: Quota of IPs on a certain network used by entities on this project
```

Comment:
I think this can be clarified a bit, how about:

> Quota of IPv4 addresses from a specified uplink network that can be used by entities in this project

The key bits I changed were changing "on" to "from" to avoid confusion around where the quota was counting resources from.

Also changed "on this project" to "in this project", as resources go "into" a project not "onto" them.